modules/s3/src/xtdb/s3/checkpoint.clj

```diff
@@ -86,10 +149,14 @@
                                        :doc "S3 bucket"}
                               :prefix {:required? false,
                                        :spec ::s3/prefix
-                                       :doc "S3 prefix"}}}
-  [{:keys [^S3Configurator configurator bucket prefix]}]
+                                       :doc "S3 prefix"}
+                              :transfer-manager? {:required? false,
```

Comment:
one to discuss: is it worth us having two modules for this, rather than switching on this flag in a number of places?

e.g. so that a user would configure it as
```clojure
:checkpointer
{:xtdb/module `xtdb.checkpoint/->checkpointer
 :store {:xtdb/module `xtdb.s3.checkpoint/->tfrmgr-cp-store
         :bucket "checkpoint-bucket"
         :prefix "checkpoint-dir"
         :configurator `s3-configurator}
 :approx-frequency (java.time.Duration/ofSeconds 30)}
```